src/vs/workbench/parts/preferences/electron-browser/settingsEditor2.ts

```diff
@@ -248,6 +241,7 @@ export class SettingsEditor2 extends BaseEditor {
 
 	clearInput(): void {
 		this.inSettingsEditorContextKey.set(false);
+		this.editorMemento.clearState(this.input, this.group);
```

Comment:
If we clean the state just before clearing input, it works, and has basically the same effect (clear the state when settings are invisible)